ui/pages/destinations/add.js

```diff
@@ -0,0 +1,181 @@
+  const [name, setName] = useState('')
+  const [connectorFullName, setConnectorFullName] = useState('')
+  const [currentDestinationName, setCurrentDestinationName] = useState('')
+  const [connected, setConnected] = useState(false)
+  const [enabledCommandInput, setEnabledCommandInput] = useState(false)
+  const [numDestinations, setNumDestinations] = useState(0)
+
+  useEffect(() => {
+    const handleDestinationConnection = () => {
+      if (accessKey && name.length > 0) {
+        fetch(`/v1/destinations?name=${connectorFullName}`)
+          .then((response) => response.json())
+          .then((data) => {
+            if (!connected) {
+              if (data.length === numDestinations) {
+                pollingTimeout = setTimeout(handleDestinationConnection, 5000)
+              } else {
+                setConnected(true)
+                clearTimeout(pollingTimeout)
+              }
+            }
+          })
+          .catch((error) => {
+            console.error(error)
+            clearTimeout(pollingTimeout)
+          })
+      }
+    }
+
+    let pollingTimeout = setTimeout(handleDestinationConnection, 5000)
+
+    return () => {
+      clearTimeout(pollingTimeout)
+    }
+  }, [accessKey])
+
+  const handleFinished = () => {
+    Router.replace('/destinations')
+  }
+
+  const handleNext = () => {
+    const type = 'kubernetes'
+    const destinationName = type + '.' + name
+
+    setCurrentDestinationName(name)
+    setEnabledCommandInput(name.length > 0)
+    setConnectorFullName(destinationName)
+    setNumDestinations(destinations.filter((item) => item.name === name).length)
+
+    fetch('/v1/identities?name=connector')
+      .then((response) => response.json())
+      .then((data) => {
+        const { id } = data[0]
+        const keyName = name + '-' + [...Array(10)].map(() => (~~(Math.random() * 36)).toString(36)).join('')
+
+        return { identityID: id, name: keyName, ttl: '87600h', extensionDeadline: '720h' }
+      })
+      .then((config) => {
+        return fetch('/v1/access-keys', {
+          method: 'POST',
+          body: JSON.stringify(config)
+        })
+      })
+      .then((response) => response.json())
+      .then((accessKeyInfo) => {
+        setAccessKey(accessKeyInfo.accessKey)
+      })
+      .catch((error) => console.error(error))
+  }
+
+  const handleKeyDownEvent = (key) => {
+    if (key === 'Enter' && name.length > 0) {
+      handleNext()
+    }
+  }
+
+  return (
+    <Fullscreen closeHref='/destinations' verticalCenteredContent={false}>
+      <Head>
+        <title>Infra - Destinations</title>
+      </Head>
+      <div className='flex flex-col mb-10 w-full max-w-md'>
+        <HeaderIcon iconPath='/destinations-color.svg' position='center' />
+        <h1 className='text-xl font-bold tracking-tight text-center'>Connect a Cluster</h1>
+        <div className='flex gap-1 mt-8 mb-5'>
+          <div className='flex-1 w-full'>
+            <InputDropdown
```

Comment:
What's the advantage of using an `InputDropdown` here vs simply an `<input>`?

Reply:
The input dropdown comes with the dropdown inside the input (the one in the share dialog on destination page). The reason why I turned it into a component is because I think this design pattern will be used in other area of the ui. However, we don't have that right now in the ui just yet. So it doesn't really need to be a component